src/ball.py

```diff
@@ -25,29 +29,185 @@ def draw(self, screen, pygame):
         # cast x and y to int for drawing
         pygame.draw.circle(screen, self.color, [int(self.position.x), int(self.position.y)], self.radius)
 
-# class BouncingBall(???):
-#     """
-#     ball effected by gravity
-#     """
-#     # TODO: 
+class BouncingBall(Ball):
+    """
+    ball effected by gravity
+    """
+    def __init__(self, bounds, position, velocity, color, radius):
+        #big: f(100 * x = 0 , ((100 - 20) / 100) * -1 = -0.2
+        #small: f(0.1) = 1 , ((100 - 10) / 100) * -1 = 
+        self.g = -1 * ((100 - radius) / 100)
+        self.friction = -1
+        super().__init__(bounds, position, velocity, color, radius)
 
-# class RainbowBall(???):
-#     """
-#     Ball that changes colors
-#     """
-#     # TODO:
+    def update(self):
+        # bounce at edges.  TODO: Fix sticky edges
 
-# class BouncingRainbow(???):
-#     """
-#     Ball that changes color and is affected by gravity
-#     """
-#     # TODO:
+        # gravity affecting acceleration
+        if self.g < 0.2 + self.radius:
+            self.velocity.y += 0.2
+        
+        # reduces x direction velocity with friction
+        if self.position.x < 0 + self.radius or self.position.x > self.bounds[0] - self.radius: # screen width
+            #self.velocity.x *= -1
+            if self.friction < -1.3877787807814457e-16:
+                self.friction += 0.1
+            else:
+                self.friction = 0
+            self.velocity.x *= self.friction
 
-# class KineticBall(???):
-#     """
-#     A ball that collides with other collidable balls using simple elastic circle collision
-#     """
-#     # TODO:
+        # reduces y direction velocity with
+        if self.position.y < 0 + self.radius or self.position.y > self.bounds[1] - self.radius: # screen height
+            if self.friction < -1.3877787807814457e-16:
+                self.friction += 0.1
+            else:
+                self.velocity.x = 0
+            
+            if self.g < 0.2 + self.radius:
+                self.velocity.y = self.velocity.y * self.g
+
+            if self.g < -1.3877787807814457e-16:
+                self.g += 0.1
+
+                
+        if self.position.y > self.bounds[1] - self.radius:
+            self.position.y = self.bounds[1] - self.radius
+        
+        self.position += self.velocity
+
+
+class RainbowBall(Ball):
```

Comment:
Perhaps refactor to implement inheritance properly. And change only the code that is different.